MCUboot update breaks compilation for boards without CONFIG_WATCHDOG=y

**Describe the bug**
The MCUboot update in cfc96eaacceeaa09bb19d8df35af7bdbbf7af189 breaks compilation for at least the `twr_ke18f` board (and likely all other boards having a DTS `watchdog0` alias with status `okay` but not setting `CONFIG_WATCHDOG=y`).

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -b twr_ke18f ../bootloader/mcuboot/boot/zephyr/`
2. See error:

```
[ 95%] Linking C executable zephyr_pre0.elf
/opt/zephyr-sdk/0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: ../app/libapp.a(main.c.obj): in function `main':
/home/hebad/Projects/zephyrproject/bootloader/mcuboot/boot/zephyr/main.c:612: undefined reference to `__device_dts_ord_105'
/opt/zephyr-sdk/0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: ../app/libapp.a(loader.c.obj): in function `boot_copy_region':
/home/hebad/Projects/zephyrproject/bootloader/mcuboot/boot/bootutil/src/loader.c:1035: undefined reference to `__device_dts_ord_105'
collect2: error: ld returned 1 exit status
make[2]: *** [zephyr/CMakeFiles/zephyr_pre0.dir/build.make:120: zephyr/zephyr_pre0.elf] Error 1
make[1]: *** [CMakeFiles/Makefile2:2921: zephyr/CMakeFiles/zephyr_pre0.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
FATAL ERROR: command exited with status 2: /home/hebad/Projects/zephyrproject/venv/bin/cmake --build /home/hebad/Projects/zephyrproject/zephyr/build
```

**Expected behavior**
MCUboot compiles without errors.

**Impact**
Unable to build MCUboot for affected boards.

**Environment (please complete the following information):**
 - OS: Ubuntu Linux 20.04 LTS
 - Toolchain: Zephyr SDK v0.15.0
 - Commit SHA: a5ce3da6ab425d104d73d34f47352ae721c5d216

--- a/west.yml
+++ b/west.yml
@@ -182,7 +182,7 @@ manifest:
       groups:
         - crypto
     - name: mcuboot
-      revision: 2fcae1c1a05fe3a0892ba3c649f93cf3e1077949
+      revision: 13f63976bca672ee018f9d55f1e31f02f4135b64
       path: bootloader/mcuboot
     - name: mipi-sys-t
       path: modules/debug/mipi-sys-t

